components/qr.js

```diff
@@ -1,18 +1,23 @@
 import QRCode from 'qrcode.react'
 import { CopyInput, InputSkeleton } from './form'
 import InvoiceStatus from './invoice-status'
-import { requestProvider } from 'webln'
 import { useEffect } from 'react'
+import { useWebLN } from './webln'
 
 export default function Qr ({ asIs, value, webLn, statusVariant, description, status }) {
   const qrValue = asIs ? value : 'lightning:' + value.toUpperCase()
+  const { provider } = useWebLN()
```

Comment:
You use `useWebLN()` and also `webLn` is being passed in. Are they both needed? Note: Bitcoin Connect does not inject webln into the window by default.

Reply:
(and this won't work if `webLn` is undefined)

Reply:
This isn't relying on `window.webln`. It just confusing naming. We manually pass in a boolean to tell if we should even try to pay the invoice via a WebLN provider. It's set to `false` if we already know the invoice failed (expired, for example). 

See [_components/invoice.js_](https://github.com/stackernews/stacker.news/blob/e20adaebdc617d24958723cf85ba537b64eb8447/components/invoice.js#L17-L51):

```javascript
let webLn = true
if (invoice.confirmedAt || (invoice.isHeld && invoice.satsReceived && !expired)) {
  variant = 'confirmed'
  status = `${numWithUnits(invoice.satsReceived, { abbreviate: false })} ${successVerb || 'deposited'}`
  webLn = false
} else if (expired) {
  variant = 'failed'
  status = 'expired'
  webLn = false
} else if (invoice.cancelled) {
  variant = 'failed'
  status = 'cancelled'
  webLn = false
}

```

Reply:
> You use `useWebLN()` and also `webLn` is being passed in. Are they both needed? Note: Bitcoin Connect does not inject webln into the window by default.

Oh, didn't see the `webLn` getting passed in, thanks! Will check if that's still needed.

> (and this won't work if webLn is undefined)

Yep, but we don't rely in `window.webln` here. It's just confusing naming. We're manually passing in boolean values. See [_components/invoice.js_](https://github.com/stackernews/stacker.news/blob/e20adaebdc617d24958723cf85ba537b64eb8447/components/invoice.js#L22):

```javascript
let webLn = true
if (invoice.confirmedAt || (invoice.isHeld && invoice.satsReceived && !expired)) {
  variant = 'confirmed'
  status = `${numWithUnits(invoice.satsReceived, { abbreviate: false })} ${successVerb || 'deposited'}`
  webLn = false
} else if (expired) {
  variant = 'failed'
  status = 'expired'
  webLn = false
} else if (invoice.cancelled) {
  variant = 'failed'
  status = 'cancelled'
  webLn = false
}
```

Reply:
Ah ok, thanks for clarifying!